libcaf_test/src/test/registry.cpp

```diff
@@ -22,8 +31,25 @@ registry::suites_map registry::suites() {
   return result;
 }
 
-std::unique_ptr<factory> registry::head_;
+ptrdiff_t registry::add(factory* new_factory) {
+  if (head_ == nullptr) {
+    head_ = new_factory;
+    tail_ = head_;
+  } else {
+    tail_->next_ = new_factory;
+    tail_ = new_factory;
+  }
```

Comment:
Just a nitpick, but maybe you will like this simplification

```suggestion
  if (head_ == nullptr)
    head_ = new_factory;
  else
    tail_->next_ = new_factory;
  tail_ = new_factory;
```